src/System.Private.Reflection.Core/src/System.Private.Reflection.Core.csproj

```diff
@@ -32,15 +30,24 @@
     <ReferencePath Include="$(AotPackageReferencePath)\System.Runtime.dll" />
     <ReferencePath Include="$(AotPackageReferencePath)\System.Diagnostics.Tracing.dll" />
     <ReferencePath Include="$(AotPackageReferencePath)\System.Resources.ResourceManager.dll" />
+    <ReferencePath Include="$(AotPackageReferencePath)\System.Collections.Immutable.dll" />
```

Comment:
No, I want to include the reference in all cases to avoid losing the various weirdnesses around making the source compile in the presence of both metadata dlls. As it is written today, the compiler will not actually generate a reference to System.Reflection.Metadata, but it will force fully qualifying things that need to be fully qualified.